load.c

```diff
@@ -1532,10 +1532,16 @@ rb_f_autoload(VALUE obj, VALUE sym, VALUE file)
  *     autoload?(name, inherit=true)   -> String or nil
  *
  *  Returns _filename_ to be loaded if _name_ is registered as
- *  +autoload+.
+ *  +autoload+ in the current namespace or one of its ancestors.
  *
  *     autoload(:B, "b")
  *     autoload?(:B)            #=> "b"
+ *
+ *     module C
+ *       autoload(:D, "d")
+ *       autoload?(:D)          #=> "c"
```

Comment:
The `"c"` should be `"d"`.

Additionally, this doesn't show an example for classes, only for modules.  I recommend adding a class example after the module example:

```ruby
class E
  autoload(:F, "f")
  autoload?(:F)      #=> "f"
  autoload?(:B)      #=> "b"
end
```

Reply:
Good idea. Added!